astropy/cosmology/core.py

```diff
@@ -379,10 +384,10 @@ def __eq__(self, other: Any, /) -> bool:
             # check all parameters in 'other' match those in 'self' and 'other'
             # has no extra parameters (latter part should never happen b/c same
             # class) TODO! element-wise when there are array cosmologies
-            and set(self.__all_parameters__) == set(other.__all_parameters__)
+            and self._parameters_all == other._parameters_all
             and all(
                 np.all(getattr(self, k) == getattr(other, k))
```

Comment:
See above for just comparing `_parameters`

Reply:
SGTM.

Reply:
See above 😆